nightingale_navigation/src/room_runner.py

```diff
@@ -0,0 +1,288 @@
+            rospy.loginfo(
+                "Nav to Bedside Completed Successfully :)", logger_name=self.logger_name
+            )
+            rospy.loginfo("RoomRunner Action complete :)", logger_name=self.logger_name)
+            result.error_code = self.ErrorCode.SUCCESS.value
+            self.server.set_succeeded(result)
+            return
+        elif self.client.get_state() == GoalStatus.PREEMPTED:
+            rospy.logwarn(
+                "Nav to Room preempted by caller", logger_name=self.logger_name
+            )
+            return
+        else:
+            rospy.logerr("Failed to Nav to Bedside", logger_name=self.logger_name)
+            result.error_code = self.ErrorCode.NAV_TO_BEDSIDE_FAILED.value
+            # the action call succeeded but the actual action performance failed
+            # this is why we call set_succeeded
+            self.server.set_succeeded(result)
+            return
+
+    def preempt_cb(self):
+        rospy.logwarn("Cancelling room_runner actions", logger_name=self.logger_name)
+        self.client.cancel_all_goals()
+        result = RoomRunnerResult()
+        result.error_code = self.ErrorCode.PREEMPTED_BY_CALLER.value
+        self.server.set_preempted(result)
+
+    def generate_goal_info(self, pose, phase):
+        msg = MoveBaseGoal()
+        msg.target_pose = pose
+
+        def fb_cb(fb):
+            target = np.array([pose.pose.position.x, pose.pose.position.y])
+            cur = np.array(
+                [fb.base_position.pose.position.x, fb.base_position.pose.position.y]
+            )
+            euclidean_distance = np.linalg.norm(target - cur)
+            ret = RoomRunnerFeedback()
+            ret.euclidean_distance_to_goal = euclidean_distance
+            ret.current_phase = phase
+            self.server.publish_feedback(ret)
+
+        a_cb = lambda: rospy.loginfo(
+            "Move Base activated the action", logger_name=self.logger_name
+        )
+        return msg, fb_cb, a_cb
+
+    def publish_initial_pose(self, x, y, yaw):
+        msg = PoseWithCovarianceStamped()
+        msg.header.stamp = rospy.Time.now()
+        msg.header.frame_id = "map"
+        msg.pose.pose.position.x = x
+        msg.pose.pose.position.y = y
+        quat = tf_conversions.transformations.quaternion_from_euler(0, 0, yaw)
+        msg.pose.pose.orientation.x = quat[0]
+        msg.pose.pose.orientation.y = quat[1]
+        msg.pose.pose.orientation.z = quat[2]
+        msg.pose.pose.orientation.w = quat[3]
+        msg.pose.covariance = self.INITAL_COV.copy()
+
+        pub = rospy.Publisher("/initialpose", PoseWithCovarianceStamped, queue_size=10)
+        # Fast rate for checking if amcl is up
+        rate = rospy.Rate(100)
+        while pub.get_num_connections() == 0:
+            rospy.loginfo_throttle(5, "Waiting for amcl to subscribe to /initialpose")
+            rate.sleep()
+        pub.publish(msg)
+
+    def call_door_lookup(self, name):
```

Comment:
What will this function do? Will it find all the known positions of doors? Or is it a function to engage the movement through a door

Reply:
I see, what's  wrong with the pose being hard coded? If we are able to get the same start pose for entering the doorways wouldn't that make things easier to traverse through?

Reply:
oh I misunderstood what you meant by hardcoded here